### DISCUSSION TITLE
ESP32 WROOM-DA WITH TWO ANTENNAS

### DISCUSSION THREAD

**Discussion by Telepresent** [timmbogner/Farm-Data-Relay-System#22] [Ideas]
Hi, I would like to know if it would be possible to use a single ESP32 WROOM-DA with two antennas as a repeater.
https://hackaday.com/2022/04/04/esp32-is-two-better-than-one/

Regards

**Comment by timmbogner** [comment]
Do you mean to essentially listen with one antenna and repeat the signal with the second?
It sounds possible!
I was also inspired by Andreas's video and will be investigating the possibilities with this module. I'm thinking of a way for sensors to ping gateways and choose the best antenna based on RSSI. 

I'll keep this open until I have done some more research!

**Comment by Telepresent** [comment]
Yes, I also think its possible!

Regards

**Comment by prairietech** [comment]
The 2 antennas detailed by Andreas on the ESP32 module are used as diversity antennas.  Diversity is the ability of the receiver to vote between the 2 antennas to determine which is receiving the signal better. If you've been listening to a radio station in your car and when you stop at a red light, the station disappears. Rolling a few inches forward restores the signal.  If you had 2 antennas on your car and your car radio could automatically vote between the 2, that's diversity. But I think your original question was more about increasing the efficiency than diversity.

I try and order ESP32 modules with external antenna connectors. The built-in PC board antennas are fine for developmental work on the bench but horrible for long range real-world applications.  A poor antenna will break an otherwise good application.  

Panel or directional antennas are a huge advantage when using low power ESP32 modules.  Why use a small rubber-duck-type all directional antenna when what you really want is the signal blasted toward the next device?  You can't afford to waste signal going in directions you don't need (unless it's really a mesh network)

But using 2 antennas also has it's disadvantages.  If you were able to properly connect 2 antennas to 1 ESP32, each antenna will only receive 1/2 of the power.  A better solution would be a single panel antenna that will concentrate it's signal in a 120 degree pattern. This would only work if the other modules were within a 120 degree beamwidth of the panel antenna.

Short answer to a long story............with the low power of these modules, make your antenna work for you and not waste power in directions you don't want it to go.
